src/lib/model/PartiesModel.js

```diff
@@ -0,0 +1,257 @@
+                        ...parsed,
+                        currentState: this.state
+                    };
+                    resolve();
+                } catch(err) {
+                    reject(err); 
+                }
+            });
+            
+            // GET /parties request to the switch
+            const res = await requests.getParties(type, id, subId);
+            
+            logger.push({ res }).log(' RequestPartiesInformation sent to peer');
+            
+        } catch(error) {
+            logger.push(error).error('RequestPartiesInformation error');
+            cache.unsubscribe(channel, sid);
+            reject(error);
+        }
+    });
+}
+
+
+/**
+ * @name channelName
+ * @description generates the pub/sub channel name
+ * @param {string} type     - the party type
+ * @param {string} id       - the party id
+ * @param {string} [subId]  - the optional party subId
+ * @returns {string} - the pub/sub channel name
+ */
+function channelName(type, id, subId) {
+    const tokens = ['parties', type, id, subId];
+    return tokens.map(x => `${x}`).join('-');
+}
+
+
+/**
+ * @name generateKey
+ * @description generates the cache key used to store state machine
+ * @param {string} type     - the party type
+ * @param {string} id       - the party id
+ * @param {string} [subId]  - the optional party subId
+ * @returns {string} - the cache key
+ */
+function generateKey(type, id, subId) {
+    return `key-${channelName(type, id, subId)}`;
+}
+
+
+/**
+ * @name injectHandlersContext
+ * @description injects the config into state machine data, so it will be accessible to on transition notification handlers via `this.handlersContext`
+ * @param {Object} config   - config to be injected into state machine data
+ * @returns {Object}        - the altered specStateMachine
+ */
+function injectHandlersContext(config) {
+    return { 
+        ...specStateMachine,
+        data: {
+            handlersContext: {
+                config: { ...config }, // injects config property
+                requests:  new MojaloopRequests({
+                    logger: config.logger,
+                    peerEndpoint: config.peerEndpoint,
+                    alsEndpoint: config.alsEndpoint,
+                    dfspId: config.dfspId,
+                    tls: config.tls,
+                    jwsSign: config.jwsSign,
+                    jwsSignPutParties: config.jwsSignPutParties,
+                    jwsSigningKey: config.jwsSigningKey,
+                    wso2Auth: config.wso2Auth
+                })
+            }
+        }
+    };
+}
+
+
+/**
+ * @name create
+ * @description creates a new instance of state machine specified in specStateMachine ^
+ * @param {Object} data     - payload data
+ * @param {String} key      - the cache key where state machine will store the payload data after each transition
+ * @param {Object} config   - the additional configuration for transition handlers
+ */
+async function create(data, key, config) {
```

Comment:
I see you have gone with a more "functional" paradigm for this model, which is not in itself a problem. I am a little concerned that this would introduce a different pattern from the rest of the codebase as the existing models use a "class" pattern. I think it would be best to either stick to the existing pattern in the other models or refactor those to follow this pattern, although I realise that would be significant volume of work. I think it is not super fantastic to have the models implemented in two different patterns.

Reply:
I agree that having two patterns for models brings some inconsistency. I believe that this `functional` proposition could bring cleaner code. I didn't start refactoring all other models, because first I want to know your opinion about this approach. So what do you favor: `class'ish` or the `functional` approach to structure the module? (of course in javascript `class` is a `function` :-D

Reply:
@eoln I do not have a particularly strong opinion either way, but maybe because of my background with OO languages I personally slightly favour the "class" oriented approach. I am aware though that encapsulating state can have some consequences which complicate testing, like having to mock many dependencies and internal state initialisation etc... which brings its own issues...if functional style is well implemented it can eliminate some of that, but also if some basics are done well in OO approach those negatives can be overcome also. In the mojaloop ecosystem I think we have to both judge the opinion of "the community" but also offer some guidance as to what is sensible. Your code looks clean, elegant, understandable, extensible and maintainable and I would personally be happy if the other models followed this pattern also.

Reply:
@bushjames we have a plan to create more sync api endpoints @ Outgoing interface (at least three more:  GET /quotes, POST /authorizations, POST /transfer) so they can be called by thirdparty-scheme-adapter (scheme adapters tandem).  So we have to make a decision @lewisdaly: should we think about refactoring other already existing models in sdk-scheme-adapter, or simply refactor only GET /parties to use the existing approach...